### PR TITLE
More pivot things yay

### PR DESCRIPTION
**Pull request by aridavidson001**
added feedforward to PivotIOSim
removed all commented buttonBindings and commands from RobotContainer(only drive command left)

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java [hunk 1]
```diff
@@ -13,19 +14,21 @@ public class PivotIOSim implements PivotIO {
...
+  private final ArmFeedforward armFeedforward = new ArmFeedforward(armkS, armkG, armkV);
   private final ProfiledPIDController pivotController =
       new ProfiledPIDController(0, 0, 0, pivotConstraints);
   private double leaderAppliedVolts = 0.0;
   private double followerAppliedVolts = 0.0;
 
-
-
   /**
    * Updates Inputs
```
Comment (JacksonElia) [inline src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:33]: Docstring should be better

File: src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java [hunk 2]
```diff
@@ -35,14 +38,13 @@ public class PivotIOSim implements PivotIO {
   public void updateInputs(PivotIOInputs inputs) {
     pivotSim.update(0.02);
```
Comment (JacksonElia) [inline src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:39]: idk how this sim stuff works, hopefully I'll learn soon. Should this 0.02 be a constant?

File: src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java [hunk 3]
```diff
@@ -35,14 +38,13 @@ public class PivotIOSim implements PivotIO {
...
 
-    inputs.leaderPosition = (pivotSim.getAngleRads() / Math.PI * 2); // Rotations
-    inputs.leaderVelocity = (pivotSim.getVelocityRadPerSec() / Math.PI * 2); // Rotations/Sec
+    inputs.leaderPosition = (pivotSim.getAngleRads() / (Math.PI * 2)); // Rotations
```
Comment (JacksonElia) [inline src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:41]: I'm pretty sure there's a built in way to calculate rotations somewhere in wpi? @Ishan1522 dyk what it is

File: src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java [hunk 4]
```diff
@@ -56,7 +58,10 @@ public void setVoltage(double volts) {
   }
 
   @Override
-  public void setPivotAngle(double angle) {
-    pivotController.calculate(angle, pivotSim.getAngleRads());
+  public void setPivotAngle(double angleRots) {
```
Comment (JacksonElia) [inline src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:61]: a comment or two would be nice in this method but that's nitpicking -- it is really simple so I get there being none

File: src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java (hunk 2 above)
Comment (Ishan1522) [reply src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:39]: yes

Comment (Ishan1522) [reply src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:39]: is update frequency of rio

File: src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java (hunk 3 above)
Comment (Ishan1522) [reply src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:41]: yea there is one for radians to rotations: https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/util/Units.html#radiansToRotations(double) 

I had to create one for meters to rotations a bit ago bc they didn't have one 4 that.

File: src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java [hunk 5]
```diff
@@ -56,7 +58,10 @@ public void setVoltage(double volts) {
...
+    double angleRads = angleRots * Math.PI * 2;
+    double currentPivotAngleRots = (pivotSim.getAngleRads() / (Math.PI * 2));
+    double armFF = armFeedforward.calculate(angleRads, pivotController.getSetpoint().velocity);
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:64]: keep units consistent to rotations. 
If you are calculating the pid setpoint in rotations you should do the feedforward in rotations, not radians.

File: src/main/java/frc/robot/subsystems/pivot/PivotIOTalon.java [hunk 6]
```diff
@@ -50,7 +50,7 @@ public class PivotIOTalon implements PivotIO {
   private final SingleLinearInterpolator speakerAngleLookupValues;
   private final SingleLinearInterpolator passAngleLookupValues;
 
-  private double pivotTargetAngle;
+  private double pivotTargetAngleRots; // Rotations
 
   private final Slot0Configs controllerConfig = new Slot0Configs();
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotIOTalon.java:55]: do u guys `controllerConfig`, if not plz remove

File: src/main/java/frc/robot/subsystems/pivot/PivotConstants.java [hunk 7]
```diff
@@ -4,7 +4,7 @@ public class PivotConstants {
   public static final double PIVOT_MASS = 0.0;
   public static final double PIVOT_LENGTH = 0.0;
   public static final double PIVOT_GEARING = 0.0;
-
+  public static final double RIO_FREQUENCY = 0.02;
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotConstants.java:7]: This value is in seconds, so frequency may be a bit confusing

Comment (Ishan1522) [reply src/main/java/frc/robot/subsystems/pivot/PivotConstants.java:7]: This should be in HardwareConstants.

Or you can create a new SimConstants class if u wants idk

File: src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java [hunk 8]
```diff
@@ -30,38 +31,46 @@ public class PivotIOSim implements PivotIO {
...
+   * @param angleRots desired angle of shooter in rotations
+   */
   @Override
   public void setPivotAngle(double angleRots) {
-    double angleRads = angleRots * Math.PI * 2;
-    double currentPivotAngleRots = (pivotSim.getAngleRads() / (Math.PI * 2));
-    double armFF = armFeedforward.calculate(angleRads, pivotController.getSetpoint().velocity);
+    double currentPivotAngleRots = Units.radiansToRotations(pivotSim.getAngleRads());
+    double armFF = armFeedforward.calculate(angleRots, pivotController.getSetpoint().velocity);
     setVoltage(pivotController.calculate(angleRots, (currentPivotAngleRots + armFF)));
```
Comment (Ishan1522) [inline src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java:74]: `armFF` should be added to the `pivotController.calculate()` not the `currentPivotAngleRots`.

Comment (aridavidson001) [reply src/main/java/frc/robot/subsystems/pivot/PivotIOSim.java]: done

File: src/main/java/frc/robot/subsystems/pivot/PivotConstants.java (hunk 7 above)
Comment (aridavidson001) [reply src/main/java/frc/robot/subsystems/pivot/PivotConstants.java]: i added it to hardware constants

**Review by Ishan1522 — APPROVED** [review]
lgtm

**Comment by Ishan1522** [comment]
also, id rename PivotIOTalon to PivotIOTalonFX, just bcs thats how weve been naming everything else

**Pull request merged**